pyk/src/pyk/__main__.py

```diff
@@ -427,5 +427,44 @@ def exec_parse_outer(options: ParseOuterOptions) -> None:
         sys.stdout.write(f'{result_text}\n')
 
 
+def exec_kompilex(options: KompileXCommandOptions) -> None:
+    definition_file = Path(options.main_file).resolve()
+    search_paths = [definition_file.parent, *options.includes]
+    main_module_name = options.main_module or definition_file.stem.upper()
+
+    try:
+        final_definition = parse_outer(definition_file, main_module_name, search_paths, options.md_selector)
+    except Exception as e:
+        _LOGGER.critical(e)
+        exit(1)
+
+    if options.pre_parsed_prelude:
+        prelude_json = json.loads(options.pre_parsed_prelude.read())
+        prelude_modules = tuple(KFlatModule.from_dict(mod) for mod in prelude_json)
+        final_definition = final_definition.let(all_modules=final_definition.all_modules + prelude_modules)
+
+    syntax_module_name = options.syntax_module or main_module_name + '-SYNTAX'
+    if syntax_module_name not in [m.name for m in final_definition.all_modules]:
+        _LOGGER.warn(
+            f'Could not find main syntax module with name {syntax_module_name} in definition. Use --syntax-module to specify one. Using {main_module_name} as default.'
+        )
```

Comment:
If the missing `syntax_module_name` comes from `options`, that should be an error: #4153